### PR TITLE
EDGDEMATIC-88 Edge-dematic module not started properly.

### PR DESCRIPTION
**Pull request by pavankumar181**
Resolves [EDGDEMATIC-88](https://issues.folio.org/browse/EDGDEMATIC-88)

## Purpose
The purpose is this PR is to start the module properly without any errors being raised. 

## Approach
In post construct method createIntegrationFlows() we have some logic that's why the module is not getting started properly.
So, we have tried adding try-catch block around it. Now we can see the module is getting started.

## Evidence
MicrosoftTeams-image (8)

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [ ] Code coverage on new code is 80% or greater
   - [ ] Duplications on new code is 3% or less
   - [ ] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [ ] There are no breaking changes in this PR.
   - [ ] Check logging

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java [hunk 1]
```diff
@@ -31,6 +32,7 @@
 
 @Service
 @RequiredArgsConstructor
+@Slf4j
 public class StagingDirectorIntegrationService {
   private static final Logger LOGGER = LoggerFactory.getLogger(StagingDirectorIntegrationService.class);
```
Comment (Arghya22) [inline src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java:38]: no need to use Loggerfactory , use @Log4j2 and use logger.error / info directly

File: src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java [hunk 2]
```diff
@@ -50,13 +52,18 @@ public class StagingDirectorIntegrationService {
...
+      removeExistingFlows();
+      var tenantsUsersMap = sms.getStagingDirectorTenantsUsers();
+      for (String tenantId : tenantsUsersMap.keySet()) {
+        for (Configuration configuration : remoteStorageService.getStagingDirectorConfigurations(tenantId, sms.getStagingDirectorConnectionParameters(tenantId).getOkapiToken())) {
+          createFlows(configuration);
+        }
       }
     }
+    catch( Exception ex) {
+      log.info("createIntegrationFlows:: exception : {}, message : {}", ex, ex.getMessage());
```
Comment (Arghya22) [inline src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java:65]: should be error level

**Review by Arghya22 — CHANGES_REQUESTED** [review]
Please check below comments. Also please increase the sonar coverage

File: src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java (hunk 1 above)
Comment (pavankumar181) [reply src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java:39]: changed to @log4j2

File: src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java (hunk 2 above)
Comment (pavankumar181) [reply src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java]: changed the log level

File: src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java [hunk 3]
```diff
@@ -31,6 +32,7 @@
 
 @Service
 @RequiredArgsConstructor
+@Slf4j
```
[the remark this answers is not in the document]
Comment (pavankumar181) [reply src/main/java/org/folio/ed/service/StagingDirectorIntegrationService.java]: Added @Log4j2

**Review by gurleenkaurbp — APPROVED** [review]

**Review by nirmalr-shah — APPROVED** [review]

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**